### DISCUSSION TITLE
New Feature: add validation rule for username in framework

### DISCUSSION THREAD

**Discussion by smarcelloc** [laravel/framework#48411] [Ideas]
When exploring the Laravel ecosystem, the robustness of the validation rules already available is evident. However, I see a opportunity for improvement: the inclusion of a specific validation rule for field username. In my experience at the company I work for, where user management is essential, we often ourselves creating custom rules to validate the username field. Not only my company where I work, other systems use username like github in page register. I believe that the user validation rule can add value to the Laravel community.

### Functionality Details:

- The username can contain letters from A to Z (upper or lower case), numbers, dash (-), underscore (_) and period (.).
- Additionally, it must start with letters and numbers.

#### Examples
Valid:

- john_doe123
- alice.smith
- 1johndoe

Invalid:
- _john_doe (does not start with letter or number)
- john@doe (contains an invalid '@' character)

**Comment by shaedrich** [comment]
I doubt, there is a universal understanding when it comes down to the specific specification of a "username". Different systems would have different patterns and requirements for their usernames.

**Reply by allanmcarvalho** [reply]
But about this we can do like the password strength validator, so, we can customize it's behavior

**Reply by shaedrich** [reply]
Maybe, functionality can be extracted into a more generic `PatternRule` or the like 🤔
